src/Arcus.WebApi.Telemetry.Serilog/Correlation/CorrelationInfoEnricher.cs

```diff
@@ -1,54 +1,39 @@
 ﻿using System;
 using Arcus.WebApi.Correlation;
 using GuardNet;
-using Microsoft.Extensions.DependencyInjection;
-using Serilog.Core;
-using Serilog.Events;
+using Microsoft.AspNetCore.Http;
+using Serilog;
 
 namespace Arcus.WebApi.Telemetry.Serilog.Correlation
 {
     /// <summary>
     /// Enriches the log events with the correlation information.
     /// </summary>
-    public class CorrelationInfoEnricher : ILogEventEnricher
+    public class CorrelationInfoEnricher : IDiagnosticEnricher
```

Comment:
This is ok, but you're limiting it to HTTP requests, what about my traces?

Reply:
It's probably got to be somewhere else.

Reply:
Why can't we use the `ILogEventEnricher` which covers all telemetry?

Reply:
The request logging has the `IDiagnosticContext` and `HttpContext` as input parameters, the othe `ILogEventEnricher` has a `LogEvent` as input parameter.

Reply:
That's right but it doesn't cover everything, so we have to use `ILogEventEnricher` to support non-HTTP request telemetry as well

Reply:
So how should approach this? Like implementing the interface but not use them in the request logging?

Reply:
How about using `ILogEventEnricher` and use `services.GetService<CorrelationInfo>` for every trace?

I think that's the only way to get it working but it covers all scenarios, at least for now

Reply:
Just FYI - `ILogEventEnricher` will be called for every single message that is being written

Reply:
Yes, but than the only question is to get hold of the service provider since it's being set further up. It would require us to step away from the Serilog recommended approach of registering the logger.

Reply:
That's why it's a recommendation ;)

There is no other way so we have to go there.

Reply:
Let me try something...

Reply:
Now it uses the Serilog `ILogEventEnricher` but uses the `HttpCorrelationInfo` because that's available in the services. The `CorrelationInfo` is added as a request feature during the receival of the request.

Reply:
Nice, thanks!